src/plugins/formulas/formulas.js

```diff
@@ -382,9 +396,11 @@ export class Formulas extends BasePlugin {
    * @returns {boolean}
    */
   isFormulaCellType(row, column, sheet = this.sheetId) {
-    const cellType = this.getCellType(row, column, sheet);
-
-    return cellType === 'FORMULA' || cellType === 'MATRIX';
+    return this.engine.doesCellHaveFormula({
+      sheet,
+      row: this.hot.toPhysicalRow(row) ?? Infinity,
+      col: this.hot.toPhysicalColumn(column) ?? Infinity
```

Comment:
Same here

Reply:
The `Infinity` mentions should be removed after https://github.com/handsontable/handsontable/pull/8669/commits/d96ae619e5822d099e6120b57403f4785d3c831f.